### PR TITLE
Group CLI output by file for multi-path runs

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- group CLI results by file to avoid interleaved output when validating multiple paths
- document file-level start and end markers
- test that multi-file runs delimit output per file

## Testing
- `ruff check nixie/cli.py tests/integration/test_cli_behavior.py`
- `pyright`
- `markdownlint README.md`
- `PYTHONPATH=. pytest -q`

------
https://chatgpt.com/codex/tasks/task_e_688fd44b5c088322ab82f42e5b91e519

## Summary by Sourcery

Group CLI output by file with start and end markers, update documentation, and add an integration test for boundary markers

Enhancements:
- Group CLI results per file with start and end markers to avoid interleaved output during multi-file runs
- Refactor main to process and report on each file sequentially

Documentation:
- Document file-level start and end markers in the README

Tests:
- Add integration test to verify file-level boundary markers in CLI output

### CODE REVIEW

File: tests/integration/test_cli_behavior.py [hunk 1]
```diff
@@ -88,3 +88,25 @@ async def test_cli_behavior(
...
+    captured = capsys.readouterr()
+
+    assert exit_code == 0
+    expected_lines = [
+        f"==> {file_a}",
+        f"<== {file_a}",
+        f"==> {file_b}",
+        f"<== {file_b}",
+    ]
+    assert captured.out.splitlines() == expected_lines
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/integration/test_cli_behavior.py]: Has this now been resolved?

File: tests/integration/test_cli_behavior.py [hunk 2]
```diff
@@ -88,3 +88,25 @@ async def test_cli_behavior(
...
+
+
+@pytest.mark.asyncio
+async def test_cli_marks_file_boundaries(
+    tmp_path: Path, stub_render: AsyncMock, capsys: pytest.CaptureFixture[str]
+) -> None:
+    file_a = tmp_path / "a.md"
+    file_b = tmp_path / "b.md"
+    file_a.write_text("```mermaid\nA-->B\n```")
+    file_b.write_text("No diagrams here")
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/integration/test_cli_behavior.py:100]: Has this now been resolved?
